Update email package dependencies

StorytellerCZ opened the pull request:
Updated email packages to their latest feature releases. `nodemailer` is now on version 7. Looking at the changelog it should be safe to update, but one step at a time.

StorytellerCZ committed:
Update email package dependencies

italojs reviewed (dismissed):
doenst Npm.depends support ^ statement?
ps: travis ci is failing

StorytellerCZ committed:
Merge branch 'release-3.4.1' into feature/update-nodemailer-6.10.1

StorytellerCZ commented:
@italojs 
`doenst Npm.depends support ^ statement?` 
Don't know, but that is not what we want in packages.
Looking into the failing Travis.

StorytellerCZ committed:
Merge branch 'release-3.4.1' into feature/update-nodemailer-6.10.1

StorytellerCZ committed:
Merge branch 'release-3.4.1' into feature/update-nodemailer-6.10.1

StorytellerCZ committed:
Fix the email tests

nodemailer 6.10.1 changed the MIME header ordering — Content-Type is now emitted as the last header (just before the body separator \r\n\r\n) instead of the first.

StorytellerCZ commented:
@italojs it looks like there has been a cosmetic change in NodeMailer v6.9.11, that moved Content Type to the bottom, so I'm fixing the Content-Type in the tests and see if that helps. There is one test that I'm unsure about where to move one header, so it will be a bit of trial and error.

https://github.com/nodemailer/nodemailer/releases/tag/v6.9.11
`Ensure that Content-type is the bottom header`

StorytellerCZ commented:
On another node NodeMailer is now on version 8, so I'll try to advance us to that if test go green.

italojs commented:
> Don't know, but that is not what we want in packages.

yeah, i saw it in the packages but i'm not sure if i'm 100% agree with it, we have many PRs just to bumpup minor versions, an option is to use `~`, that bumpup just the patch version

StorytellerCZ commented:
@italojs this goes all the way back why Meteor is tied to specific version of Node. We want that control and stability, not to mention we just saw what a bump did to the test. You have to start making test matrixes for NPM packages and further complicate the test suite. Plus, if I understand things correctly, it would take only the current version of the NPM package at the moment of packaging, so you will still not get any benefit from it and will have to make a new package release either way to get the latest dependency.

StorytellerCZ commented:
Either way the tests have passed, so I will try to push NodeMailer to the latest version.

StorytellerCZ committed:
NodeMailer v8.0.1

Try pushing NodeMailer to the latest version

StorytellerCZ commented:
OK, so we got green on NodeMailer v8.0.1, so it looks like it was just the order of headers.

nachocodoner commented:
I do not see a Meteor API break in this PR itself. The main upstream breaking change in Nodemailer 8 is the error code rename [from NoAuth to ENOAUTH](https://github.com/nodemailer/nodemailer/releases/tag/v8.0.0), so this should be fine unless we (I checked and seems we don't) or downstream users rely on the old exact code string.

Also, I will bump Nodemailer to latest patch 8.0.3.

nachocodoner committed:
Update nodemailer dependency version

Updated nodemailer version from 8.0.1 to 8.0.3.

nachocodoner commented:
I am reluctant to add this fix to Meteor 3.4.1. Patch releases are expected to remain backward compatible under semver, while Nodemailer 8.0.0 is a major release and explicitly includes a breaking change: NoAuth => ENOAUTH.

It is fair to say Meteor apps are unlikely to be affected once core is adapted, because Nodemailer is mostly an internal dependency behind `meteor/email`. Still, I would not claim that no users can be affected. Meteor apps that use only `Email.send` in a basic way are less likely to notice the Nodemailer bump. Meteor apps using custom transport-style integrations, hooks, attachments, or other Nodemailer-shaped options are the ones more likely to observe any upstream behavior change.

We could mention this indirect breaking change in Meteor 3.4.1, but following semver convention, I think we should avoid the risk of introducing a breaking change in a patch release.

What do you think, @italojs? Should we move it to Meteor 3.5? Also, if we move it to 3.5, we should not merge it yet, since this PR is based on the Meteor 3.4.1 state, and that is not yet available in the 3.5 branch.

italojs commented:
> @italojs this goes all the way back why Meteor is tied to a specific version of Node. We want that control and stability
got it, still dont agree with this control level since patches shouldn't affect anybody, but ok, let's maintain as it is

> What do you think, @italojs? Should we move it to Meteor 3.5? Also, if we move it to 3.5, we should not merge it yet, since this PR is based on the Meteor 3.4.1 state, and that is not yet available in the 3.5 branch.

Yes, it's good to maintain the version consistency, let's move to 3.5 (after release 3.4.1)

StorytellerCZ committed:
Merge branch 'devel' into feature/update-nodemailer-6.10.1

StorytellerCZ committed:
Merge branch 'release-3.5' into feature/update-nodemailer-6.10.1
